rclcpp/test/rclcpp/test_create_timer.cpp

```diff
@@ -193,3 +193,29 @@ TEST(TestCreateTimer, timer_function_pointer)
 
   rclcpp::shutdown();
 }
+
+TEST(TestCreateTimer, timer_without_autostart)
+{
+  rclcpp::init(0, nullptr);
+  auto node = std::make_shared<rclcpp::Node>("test_create_timer_node");
+
+  rclcpp::TimerBase::SharedPtr timer;
+  timer = rclcpp::create_timer(
+    node,
+    node->get_clock(),
+    rclcpp::Duration(0ms),
+    []() {},
+    nullptr,
+    false);
+
+  EXPECT_TRUE(timer->is_canceled());
+  EXPECT_EQ(timer->time_until_trigger().count(), std::chrono::nanoseconds::max().count());
+
+  timer->reset();
```

Comment:
I believe this needs to be well-described in doc section. user application must call `reset` the timer for starting.

besides, i would like to have feedback on https://github.com/ros2/rcl/pull/1004#pullrequestreview-1111587769

CC: @alsora @gbiggs

Reply:
I think this still needs to be addressed in doc section for user explanation.

Reply:
@Voldivh can you check this comment?

Reply:
Sure thing! Just to be clear, we are talking about adding more explanation on the .hpp file on the timer class section?